bindings/python/scripts/parse.py

```diff
@@ -96,73 +97,34 @@ def generate_parsed_info(node):
-    if cursor.raw_comment:
-        parsed_info["raw_comment"] = cursor.raw_comment
-
-    # add result of various kinds of checks available in cindex.py
-
-    cursorkind_checks = {
-        "kind_is_declaration": cursor.kind.is_declaration,
-        "kind_is_reference": cursor.kind.is_reference,
-        "kind_is_expression": cursor.kind.is_expression,
-        "kind_is_statement": cursor.kind.is_statement,
-        "kind_is_attribute": cursor.kind.is_attribute,
-        "kind_is_invalid": cursor.kind.is_invalid,
-        "kind_is_translation_unit": cursor.kind.is_translation_unit,
-        "kind_is_preprocessing": cursor.kind.is_preprocessing,
-        "kind_is_unexposed": cursor.kind.is_unexposed,
-    }
-
-    # check for deleted ctor analogous to `is_default_constructor` unavailable
-    cursor_checks = {
-        "is_definition": cursor.is_definition,
-        "is_const_method": cursor.is_const_method,
-        "is_converting_constructor": cursor.is_converting_constructor,
-        "is_copy_constructor": cursor.is_copy_constructor,
-        "is_default_constructor": cursor.is_default_constructor,
-        "is_move_constructor": cursor.is_move_constructor,
-        "is_default_method": cursor.is_default_method,
-        "is_mutable_field": cursor.is_mutable_field,
-        "is_pure_virtual_method": cursor.is_pure_virtual_method,
-        "is_static_method": cursor.is_static_method,
-        "is_virtual_method": cursor.is_virtual_method,
-        "is_abstract_record": cursor.is_abstract_record,
-        "is_scoped_enum": cursor.is_scoped_enum,
-        "is_anonymous": cursor.is_anonymous,
-        "is_bitfield": cursor.is_bitfield,
-    }
-
-    type_checks = {
-        "type_is_const_qualified": cursor.type.is_const_qualified,
-        "type_is_volatile_qualified": cursor.type.is_volatile_qualified,
-        "type_is_restrict_qualified": cursor.type.is_restrict_qualified,
-        "type_is_pod": cursor.type.is_pod,
-    }
-
-    for checks in (cursorkind_checks, cursor_checks, type_checks):
-        for check, check_call in checks.items():
-            parsed_info[check] = check_call()
-
-    # special case handling for `cursor.type.is_function_variadic()`
-    if cursor.type.kind.spelling == "FunctionProto":
-        parsed_info["type_is_function_variadic"] = cursor.type.is_function_variadic()
-
     parsed_info["members"] = []
 
+    # add result of various kinds of checks available in cindex.py via clang_utils.py
+    for key, object in (
+        ("cursor_kind", CursorKindUtils(cursor_kind=cursor.kind)),
+        ("cursor", CursorUtils(cursor=cursor)),
+        ("type", TypeUtils(cursor_type=cursor.type)),
+    ):
+        parsed_info[key] = {
+            **object.check_functions_dict,
+            **object.get_functions_dict,
+            **object.properties_dict,
+        }
+
+    # Hacky fixes
+
+    # get spelling from object
+    parsed_info["cursor"]["result_type"] = parsed_info["cursor"]["result_type"].spelling
```

Comment:
Any need to check if cursor.result_type is empty?

Reply:
https://github.com/PointCloudLibrary/clang-bind/blob/957433bd8c2b8fd985da5e77dd9661a4cace5687/bindings/python/scripts/parse.py#L110-L111

We were just checking for `""`earlier, so it won't be a problem. We removed all these custom checks in this PR.